src/python/pants/core/goals/deploy.py

```diff
@@ -83,6 +84,16 @@ class DeploySubsystem(GoalSubsystem):
     name = "experimental-deploy"
     help = "Perform a deployment process."
 
+    dry_run = BoolOption(
```

Comment:
should we display a warning or raise an error when using this option with targets that do not handle this yet? Could be annoying if you want to dry run a helm deployment which then deploys for real.

Reply:
Hmm, some deployments such as helm handle this directly with a passthrough argument. Will this in some way clash with this goal-level option? https://github.com/pantsbuild/pants/blob/main/src/python/pants/backend/helm/subsystems/helm.py#L28

Reply:
In the helm plugin `--dry-run` is set via the `--helm-args` flag so thats a different mechanism. Maybe I should not put the `--dry-run` flag on the goal level then and move it inside terraform somehow?

Reply:
Alternatively I could remove the dry run option from helms passthrough args and make helm use the goal level flag.